### PR TITLE
Add read-only REST API for site/scan data; enable CORS for GET/HEAD

### PR DESCRIPTION
**Pull request by eloquence**
This uses the Django REST framework, which adds a nice browsable API
as well.

API can be found at /api/v1, /api redirects to it.

### CODE REVIEW

**Comment by eloquence** [comment]
That all makes sense, thanks! Will poke.

**Comment by eloquence** [comment]
Okay, I think I've got the latest_scan bit working (https://github.com/freedomofpress/securethenews/pull/109/commits/08afcd4e8f14f15df7f5411fe62c15082d0a9a29), and have updated the README. Let me know if that implementation looks sane to you. I'll poke at fleshing out the site view -- my hunch is that we should keep the latest_scan field there as well, and have an all_scans field that links to a /scans view as you suggest.

File: securethenews/api/views.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+from rest_framework import routers, viewsets, generics, filters as rest_framework_filters
+from sites.models import Site
+from rest_framework.decorators import api_view
+from rest_framework.response import Response
+from rest_framework.reverse import reverse
+
+"""
+NOTE: Method docstrings in this file are used by the framework to generate
+page subtitles
+"""
```
Comment (garrettr) [inline securethenews/api/views.py]: Nit: don't use multi-line strings for comments.

File: securethenews/api/views.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+
+@api_view(['GET'])
+def api_root(request, format=None):
+    """
+    Index of available API calls
+    """
+
```
Comment (garrettr) [inline securethenews/api/views.py]: Nit: remove unnecessary newline

**Comment by eloquence** [comment]
Per-site scan lists should now work, ordered, filterable & paginated.

All files have also gone through autopep8, and I've moved module comments to the top of each file.

Let me know if that looks good. If so, do you want me to prepare a fresh single-commit PR, or are you OK cherry-picking the relevant commits?

**Comment by garrettr** [comment]
> If so, do you want me to prepare a fresh single-commit PR, or are you OK cherry-picking the relevant commits?

Don't worry about that, I'll handle that. Will probably do a squash merge. Reviewing your latest changes now.

**Comment by garrettr** [comment]
@thisisparker Since your Twitter bot is the motivating use case for the Secure the News API, care to comment on the API implemented by this pull request? Is there anything that needs to be added or changed for it to be used by the Secure the News Twitter bot?

**Comment by conorsch** [comment]
@eloquence This is great work, thank you for taking the time!

> What's missing is the score-derived grade. 

Hmm, that sounds pretty critical for @thisisparker's use case, but I'll let him chime in to confirm.

**Comment by eloquence** [comment]
It can be pretty easily calculated, but let me poke - should be easy to include so people don't have to do that themselves.

**Comment by thisisparker** [comment]
Sorry for the delay, I was traveling, but this all looks good from my end. Of course I'll have to rewrite a bit to _use_ the API, but it looks like that will be just a few lines different and I'll end up with a much more robust app. Thanks for working on this, @eloquence!

**Comment by eloquence** [comment]
Thanks for taking a look, Parker, and thanks for running the bot! Happy to help with any issues that may arise in migrating.

**Comment by garrettr** [comment]
Syncing up on this after a few days off; I think it's ready to merge, but doing a final review right now 😁

**Comment by garrettr** [comment]
This looks great, @eloquence! Thanks so much for your contribution, and for going back and forth with me on a few rounds of review. I'm going to merge this now, and will comment again on this PR when we've deployed these changes into production.

**Comment by garrettr** [comment]
@eloquence Changes are live on the production site! https://securethe.news/api/v1/ 🍾 

Thanks again for this excellent contribution! Would you mind if we wrote a quick blog post announcing the change, and credited you by your Github handle?

**Comment by eloquence** [comment]
Sweet, thanks for all the help shepherding this to production. :) +1 to a blog post, will share the news once it's up.

**Comment by thisisparker** [comment]
Finally pushing an update to the bot to make it use this API. Pretty seamless, good work everybody! :+1:

**Pull request merged**